### PR TITLE
add support for date_spine

### PR DESCRIPTION
**Pull request by TwoMinds**
added date_spine macro. Glad to be able to give back to the community. 

thanks,
-Serge

### CODE REVIEW

File: integration_tests/dbt_utils/dbt_project.yml [hunk 1]
```diff
@@ -31,9 +31,6 @@ models:
       test_datediff: *not-synapse
       test_hash: *not-synapse
       test_last_day: *not-synapse
-    datetime:
-      test_date_spine: &disabled # BROKEN DUE TO MODEL DEF
-        +enabled: false
     materializations:
       test_insert_by_period: *disabled
```
Comment (swanderz) [inline integration_tests/dbt_utils/dbt_project.yml:35]: my bad, can you make this change? it has to do w/ YAML anchors

```suggestion
      test_insert_by_period: &disabled
        +enabled: false
```

File: macros/dbt_utils/datetime/date_spine.sql [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    select * from filtered
+
+{% endmacro %}
+
+
+{% macro sqlserver__date_spine(start_date, end_date, datepart) -%}
+
+    {% set date_spine_query %}
+
+        {{sqlserver__date_spine_sql(start_date, end_date, datepart)}} order by 1
```
Comment (swanderz) [inline macros/dbt_utils/datetime/date_spine.sql:93]: try this. the "rule" for package macros is that you should always fully qualify them when using them inside other macros
```suggestion
        {{tsql_utils.sqlserver__date_spine_sql(start_date, end_date, datepart)}} order by 1
```

Comment (swanderz) [inline macros/dbt_utils/datetime/date_spine.sql:93]: also I'm not sure if `order by 1` works in T-SQL?

Comment (TwoMinds) [reply macros/dbt_utils/datetime/date_spine.sql:93]: Thanks. Good rule!

Comment (TwoMinds) [reply macros/dbt_utils/datetime/date_spine.sql]: order by 1 does work in T-SQL.

File: macros/dbt_utils/datetime/date_spine.sql [hunk 3]
```diff
@@ -0,0 +1,119 @@
+{% macro sqlserver__date_spine_sql(start_date, end_date, datepart) %}
```
Comment (swanderz) [inline macros/dbt_utils/datetime/date_spine.sql:1]: this is the failure from Azure SQL. looking at `dbt-utils`'s [date_spine](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/datetime/date_spine.sql) and [the model that's used to test `date_spine`](https://github.com/fishtown-analytics/dbt-utils/blob/master/integration_tests/models/datetime/test_date_spine.sql), it looks like we need to re-arrange the params.
>  ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Invalid parameter 1 specified for datediff. (1023) (SQLExecDirectW)')

```suggestion
{% macro sqlserver__date_spine_sql(datepart, start_date, end_date) %}
```

File: macros/dbt_utils/datetime/date_spine.sql [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        where date_{{datepart}} <= {{ end_date }}
+
+    )
+
+    select * from filtered
+
+{% endmacro %}
+
+
+{% macro sqlserver__date_spine(start_date, end_date, datepart) -%}
```
Comment (swanderz) [inline macros/dbt_utils/datetime/date_spine.sql:89]: same here

```suggestion
{% macro sqlserver__date_spine(datepart, start_date, end_date) -%}
```

File: macros/dbt_utils/datetime/date_spine.sql [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    {% set date_spine_query %}
+
+        {{tsql_utils.sqlserver__date_spine_sql(start_date, end_date, datepart)}} order by 1
```
Comment (swanderz) [inline macros/dbt_utils/datetime/date_spine.sql:93]: and here
```suggestion
        {{tsql_utils.sqlserver__date_spine_sql(datepart, start_date, end_date)}} order by 1
```

**Review by swanderz — APPROVED** [review]
thanks @TwoMinds -- I'm going to use this tomorrow!

**Comment by swanderz** [comment]
@TwoMinds last request -- can you add a CHANGELOG.MD and make it's contents look something like this?

```
# Changelog

## `0.6.4.1`

- add support for `date-spine` [#28](https://github.com/dbt-msft/tsql-utils/pull/28) thanks [@TwoMinds](https://github.com/TwoMinds)

## `0.6.4`

as of now, all 39 of dbt-utils's macros work on `dbt-sqlserver` and `dbt-synapse` except for the 10 listed below. We hope to bring these macros in a future release!

    insert_by_period materialization
    date_spine
    test_mutually_exclusive_ranges() #18
    all the url macros (get_url_host, get_url_path, get_url_host()
    generate_series()
    get_relations_by_pattern()
    get_relations_by_prefix_and_union()
    union_relations()

these 5 macros are supported for SQL Server and Azure SQL, but aren't yet officially supported right now on Synapse because of https://github.com/dbt-msft/dbt-synapse/issues/36 breaks dbt-utils's integration tests.  

    dateadd()
    datediff()
    split_part()
    last_day()
    test-hash()
```

**Pull request merged**